decidim-proposals/app/controllers/decidim/proposals/proposal_votes_controller.rb

```diff
@@ -4,13 +4,22 @@ module Decidim
   module Proposals
     # Exposes the proposal vote resource so users can vote proposals.
     class ProposalVotesController < Decidim::Proposals::ApplicationController
+      include ProposalVotesHelper
+
       before_action :authenticate_user!
       before_action :check_current_settings!
+      before_action :check_vote_limit_reached!, only: [:create]
 
       def create
-        @proposal = Proposal.where(feature: current_feature).find(params[:proposal_id])
-        @proposal.votes.create!(author: current_user)
+        proposal.votes.create!(author: current_user)
         @from_proposals_list = params[:from_proposals_list] == "true"
+        render :update_buttons_and_counters
+      end
+
+      def destroy
+        proposal.votes.where(author: current_user).first.destroy
```

Comment:
Do you care if it's the first one? If not you can just `proposal.votes.where(author: current_user).delete`